### PR TITLE
Tweak demo code

### PR DESCRIPTION
**Pull request by mit-mit**
- Refactor into multiple files
- Create two versions of the main `home` widget: one "unfinished" that we'll edit during the session, and one finished that folks can look at for the solution
- Use Dart style guide

### CODE REVIEW

File: isoimg/lib/home.dart [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      _loading = true;
+    });
+    final image = _applySepiaFilter(_image!);
+    setState(() {
+      _image = image;
+      _loading = false;
+    });
+  }
+
+  void _showImagePicker({@required sync}) async {
```
Comment (goderbauer) [inline isoimg/lib/home.dart]: nit: remove the parameter since it is no longer useful?

File: isoimg/lib/home.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            ),
+        ],
+      ),
+      persistentFooterButtons: <Widget>[
+        TextButton(
+          onPressed: !_loading && _image != null ? _applySepia : null,
+          child: const Text('Sepia'),
+        ),
+        TextButton(
+          onPressed: _image != null ? _undo : null,
```
Comment (goderbauer) [inline isoimg/lib/home.dart]: Only offer undo if there's something to undo?

```suggestion
          onPressed: _image != _originalImage ? _undo : null,
```

File: isoimg/lib/home_completed.dart [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        TextButton(
+          onPressed: !_loading && _image != null ? _applySepiaAsync : null,
+          child: const Text('Sepia (async)'),
+        ),
+        TextButton(
+          onPressed: !_loading && _image != null ? _applySepiaInIsolate : null,
+          child: const Text('Sepia (isolate)'),
+        ),
+        TextButton(
+          onPressed: !_loading && _image != null ? _undo : null,
```
Comment (goderbauer) [inline isoimg/lib/home_completed.dart:164]: same as above

File: isoimg/lib/home.dart (hunk 2 above)
Comment (goderbauer) [reply isoimg/lib/home.dart]: Here and for clear we should probably also guard with `!_loading`.

**Review by goderbauer — APPROVED** [review]
LGTM, just some nits.

File: isoimg/lib/home.dart (hunk 1 above)
Comment (mit-mit) [reply isoimg/lib/home.dart]: done

File: isoimg/lib/home.dart (hunk 2 above)
Comment (mit-mit) [reply isoimg/lib/home.dart]: done

File: isoimg/lib/home_completed.dart (hunk 3 above)
Comment (mit-mit) [reply isoimg/lib/home_completed.dart:164]: done

**Pull request merged**